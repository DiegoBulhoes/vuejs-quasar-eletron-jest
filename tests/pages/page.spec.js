import { shallowMount, createLocalVue } from '@vue/test-utils';
import pageError404 from '@/pages/Error404.vue';
import pageIndex from '@/pages/Index.vue';
import Quasar from 'quasar';

describe('Something Something', () => {
  let localVue;
  let wrapper;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Quasar);
  });

  test('Index - isVueInstance', () => {
    wrapper = shallowMount(pageIndex, {
      localVue,
    });
    expect(wrapper.isVueInstance()).toBe(true);
  });

  test('renders correctly - Index', () => {
    wrapper = shallowMount(pageIndex, {
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  test('Error404 - isVueInstance', () => {
    wrapper = shallowMount(pageError404, {
      localVue,
    });
    expect(wrapper.isVueInstance()).toBe(true);
  });

  test('renders correctly - Error404', () => {
    wrapper = shallowMount(pageError404, {
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
