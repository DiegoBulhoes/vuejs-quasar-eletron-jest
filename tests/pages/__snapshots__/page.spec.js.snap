// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Something Something renders correctly - Error404 1`] = `
<div
  class="fixed-center text-center"
>
  <p>
    <img
      src="~assets/sad.svg"
      style="width: 30vw; max-width: 150px;"
    />
  </p>
   
  <p
    class="text-faded"
  >
    Sorry, nothing here...
    <strong>
      (404)
    </strong>
  </p>
   
  <q-btn-stub
    color="secondary"
    style="width: 200px;"
  >
    Go back
  </q-btn-stub>
</div>
`;

exports[`Something Something renders correctly - Index 1`] = `
<q-page-stub
  class="flex flex-center"
>
  <img
    alt="Quasar logo"
    src="~assets/quasar-logo-full.svg"
  />
</q-page-stub>
`;
